File: Project1/14719.cpp
#include <iostream>
#include <vector>

using namespace std;
int main(void)
{
    int h, w;
    cin >> h >> w;
    vector<int> block(w);
    for (int i=0; i<w; ++i)
    {
        cin >> block[i];
    }
    int result = 0;
    // left
    // find start
    int start = 0, end = 0;
    for (int i=0; i<w; ++i)
    {
        if (block[i] > 0)
        {
            start = i;
            break;
        }
    }
    // find end
    for (int i=start+1; i<w; ++i)
    {
        if (block[i] >= block[start])
        {
            end = i;
            // fill
            for(int j=start+1; j<end; ++j)
            {
                result += block[start] - block[j];
            }
            start = end;
        }
    }

    // right
    // find start
    for (int i=w-1; i>=0; --i)
    {
        if (block[i] > 0)
        {
            start = i;
            break;
        }
    }
    // find end
    for (int i=start-1; i>=0; --i)
    {
        if (block[i] > block[start])
        {
            end = i;
            // fill
            for(int j=start-1; j>end; --j)
            {
                result += block[start] - block[j];
            }
            start = end;
        }
    }

    cout << result;
}
// 2 3
// 2 0 2 중복 계산 방지 위해 왼쪽/오른쪽 중 한 곳은 더 큰 블록으로!

File: Project1/14719.py
import sys
input = sys.stdin.readline
h, w = map(int, input().strip().split())
blocks = list(map(int, input().strip().split()))
result = 0
# left
# find start
start = 0
end = 0
for i in range(w):
    if blocks[i] > 0:
        start = i
        break
# find end
for i in range(start+1, w):
    if blocks[i] >= blocks[start]:
        end = i
        # fill
        for j in range(start+1, end):
            result += blocks[start] - blocks[j]
        start = end
# right
# find start
start = w-1
end = w-1
for i in range(w-1, -1, -1):
    if blocks[i] > 0:
        start = i
        break
# find end
for i in range(start-1, -1, -1):
    if blocks[i] > blocks[start]:
        end = i
        # fill
        for j in range(start-1, end, -1):
            result += blocks[start] - blocks[j]
        start = end
print(result)
